src/components/datasets/DatasetDataTypes.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Button, Col, Row, Table, Typography } from "antd";
+
+import PropTypes from "prop-types";
+import { datasetPropTypesShape, projectPropTypesShape } from "../../propTypes";
+import { clearDatasetDataType } from "../../modules/metadata/actions";
+import { fetchDatasetDataTypesSummaryIfPossible } from "../../modules/datasets/actions";
+import genericConfirm from "../ConfirmationModal";
+import { nop } from "../../utils/misc";
+import DataTypeSummaryModal from "./datatype/DataTypeSummaryModal";
+
+const NA_TEXT = <span style={{ color: "#999", fontStyle: "italic" }}>N/A</span>;
+
+const DatasetDataTypes = React.memo(({isPrivate, project, dataset, onDatasetIngest:
+    onDatasetIngest, isFetchingDatasets }) => {
+    const dispatch = useDispatch();
+
+    const datasetDataTypes = useSelector((state) => state.datasetDataTypes.itemsById[dataset.identifier]);
+    const datasetSummaries = useSelector((state) => state.datasetSummaries.itemsById[dataset.identifier]);
+
+    const [datatypeSummaryVisible, setDatatypeSummaryVisible] = useState(false);
+    const [selectedDataType, setSelectedDataType] = useState(null);
+
+    const selectedSummary = (selectedDataType !== null && datasetSummaries)
+        ? datasetSummaries[selectedDataType.id]
+        : {};
+
+    const handleDeleteDataType = async (dataType) => {
+        genericConfirm({
+            title: `Are you sure you want to delete the "${dataType.label || ""}" data type?`,
+            content: "Deleting this means all instances of this data type contained in the dataset " +
+                "will be deleted permanently, and will no longer be available for exploration.",
+            onOk: async () => {
+                await dispatch(clearDatasetDataType(dataset.identifier, dataType.id));
+                await dispatch(fetchDatasetDataTypesSummaryIfPossible(dataset.identifier));
+            },
+        });
+    };
+
+    const showDatatypeSummary = (dataType) => {
+        setSelectedDataType(dataType);
+        setDatatypeSummaryVisible(true);
+    };
+
+    const dataTypesColumns = [
+        {
+            title: "Name",
+            key: "label",
+            render: (dt) =>
+                isPrivate ? (
+                    <a style={{ fontFamily: "monospace" }} onClick={() => showDatatypeSummary(dt)}>
```

Comment:
don't make this monospace, since it's a human-readable label and not an ID or something.